### ISSUE TITLE
Mac command uses deprecated -K flag when setting up SSH keys

### ISSUE THREAD

**Issue by tsongas** [github/docs#13946]
### Code of Conduct

- [X] I have read and agree to the GitHub Docs project's [Code of Conduct](https://github.com/github/docs/blob/main/CODE_OF_CONDUCT.md)

### What article on docs.github.com is affected?

https://docs.github.com/en/authentication/connecting-to-github-with-ssh/generating-a-new-ssh-key-and-adding-it-to-the-ssh-agent

### What part(s) of the article would you like to see updated?

Running `ssh-add -K ~/.ssh/id_ed25519` on a Mac results in the following console warning, so both the command and the callout below it need to be updated.

```
WARNING: The -K and -A flags are deprecated and have been replaced
         by the --apple-use-keychain and --apple-load-keychain
         flags, respectively.  To suppress this warning, set the
         environment variable APPLE_SSH_ADD_BEHAVIOR as described in
         the ssh-add(1) manual page.
```

### Additional information

I'm on a new 16" MacBook Pro.

**Comment by ramyaparimi** [comment]
@tsongas 
Thanks so much for opening an issue! I'll triage this for the team to take a look :eyes:

**Comment by stevecat** [comment]
Hi @tsongas, thanks for opening this! ✨ 

When we first spotted the deprecation notice, I wanted to update all the commands in the article with the replacement flags but learnt that support for them only appeared recently (as far as I can see, they were first available in in Monterey).

Without providing separate commands for different versions of MacOS, I think we're best keeping the deprecated flags for the time being so we can be confident they will function for all users. The deprecation notice is better than the flags simply not being recognised. 

We added a callout about the deprecation notice in https://github.com/github/docs/pull/11443 and we'll continue to keep an eye on this. Once we can be confident that they will work for the vast majority of users, or Apple remove support for `-K` and `-A` entirely, we'll update the article.

I'll close this issue for now but do @mention me if you have any extra information about the change; it was a bit of a struggle to determine when support was added! 😄
